<!DOCTYPE html>
<html lang="en">
  <head>
    <title>CRISPResso2</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="CRISPResso2: Analysis of genome editing outcomes from deep sequencing data">
    <meta name="author" content="Kendell Clement and Luca Pinello">
    <link href='https://fonts.googleapis.com/css?family=Montserrat|Ranga:700' rel='stylesheet' type='text/css'>
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap/5.1.3/css/bootstrap.min.css" integrity="sha512-GQGU0fMMi238uA+a/bdWJfpUGKUkBdgfFdgBm72SUQ6BeyWjoY/ton0tEjH+OSH9iP4Dfh+7HM0I9f5eR0L/4w==" crossorigin="anonymous" referrerpolicy="no-referrer" />    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.3.5/jquery.fancybox.min.css" />
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.0.13/css/all.css" integrity="sha384-DNOHZ68U8hZfKXOrtjWvjxusGo9WQnrNx2sqG0tfsghAvtVlRW3tvkXWZh58N9jp" crossorigin="anonymous">
    <style>
      body {
        font-family: 'Montserrat', sans-serif !important;
      
      }
      @media (max-width:1015px)
      {
        #crispresso_h
        {
          font-size:5rem;
          font-weight:300;
          line-height:1.2;
        }
        #left_help_div
        {
          display:none
        }
        #top_help_div
        {
          display:block
        }
      }
      @media (min-width:1015px)
      {
        #crispresso_h
        {
          font-size:8rem;
          font-weight:300;
          line-height:1.2;
        }
        #left_help_div
        {
          display:block
        }
        #top_help_div
        {
          display:none
        }
      }
    </style>


    <!-- Always force latest IE rendering engine (even in intranet) & Chrome Frame
    Remove this if you use the .htaccess -->
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

    <!-- Replace favicon.ico & apple-touch-icon.png in the root of your domain and delete these references -->
    
    <link rel="shortcut icon" href="http://crispresso.pinellolab.org/static/favicon.ico" />
    

    <!-- Optional JavaScript -->  
    <!-- jQuery first, then Popper.js, then Bootstrap JS -->
    <script src="https://code.jquery.com/jquery-3.3.1.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.3.5/jquery.fancybox.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p" crossorigin="anonymous"></script>


  <script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+'://platform.twitter.com/widgets.js';fjs.parentNode.insertBefore(js,fjs);}}(document, 'script', 'twitter-wjs');</script>
  

<script src="https://cdn.plot.ly/plotly-2.11.1.min.js"></script>
<style>
.nav-tabs.amp-header {
  border-bottom:none !important;
}

.card-header.amp-header {
  border-bottom:none !important;
}
.nav-tabs.amp-header .nav-link.active {
  background-color:lightsteelblue;
  border-bottom:lightsteelblue;
}

.tab-content.amp-body {
  background-color:lightsteelblue;
}

@media print {
   .tab-content > .tab-pane {
    display: block !important;
    opacity: 1 !important;
    visibility: visible !important;
		margin-bottom: 2em !important;
		page-break-inside: avoid;
  }
  .nav-tabs {
    display:none !important;
    visibility:hidden !important;
  }
  .tab-content.amp-body {
    background-color:transparent !important;
  	border:None !important;
  }
}
@media only screen and (max-width: 600px) {
	.jumbotron img {
		width:100%
	}
}
</style>



  </head>

  <body class="m-2">
    

  
  
  <div class="container">
      <div class="row pb-2">
        <div class="col-sm-1"></div>
        
          <div class="col-sm-3 crispresso_cup"> <a href='https://crispresso2.pinellolab.org'><img class='img-fluid' src="http://crispresso.pinellolab.org/static/imgs/CRISPResso_justcup.png" width="80%"></a>
        
          </div>
        <div class="col-sm-7" >
          <a href="https://crispresso2.pinellolab.org" style="text-decoration:none !important; color:rgb(33, 37, 41)"><h1 id='crispresso_h' style="font-family:'Ranga', sans-serif">CRISPResso2 </h1></a>
          <h3 style="margin-top: -0.5em;padding-right: 2em;">Analysis of genome editing outcomes from deep sequencing data</h3>
        </div>
        <div class="col-sm-1">
        </div>
      </div>


    

      <div id='top_help_div'>
        <div class="row">
          <div class="col-sm-1"></div>
          <div class='col-sm-10 pb-2 crispresso_body_div'>
           
          </div>
          <div class="col-sm-1"></div>
        </div>
      </div>

      
<div class="row">
<div class="col-sm-1"></div>
<div class="col-sm-10">

	<div class="jumbotron" style="background:rgba(0,0,0,0.0); padding:0px" >
		<div id='jumbotron_content' >

          

            

          

          

          

          

          
            
            <div class='card text-center mb-2'>
              <div class='card-header'>
                <h5>Editing efficiency comparison</h5>
              </div>
              <div class='card-body' style='max-height:80vh;overflow-y:auto'>
		        <div id="fig_summary_1.Editing_comparison">
    
        
            <a href="1.Editing_comparison.pdf"><img src="1.Editing_comparison.png" width='70%'></a>
        
    
    <label class="labelpadding">Figure 1: Comparison for amplicon Reference; Left: Percentage of modified and unmodified reads in each sample; Right: relative percentage of modified and unmodified reads</label>
    
        <p class="m-0"><small>Data: <a href="../CRISPRessoBatch_on_FANC/CRISPResso_on_Cas9/Cas9.CRISPResso2_report.html">Cas9 output</a></small></p>
    
        <p class="m-0"><small>Data: <a href="../CRISPRessoBatch_on_FANC/CRISPResso_on_Untreated/Untreated.CRISPResso2_report.html">Untreated output</a></small></p>
    
    <br>
</div>
              </div>
            </div>
            
            <div class='card text-center mb-2'>
              <div class='card-header'>
                <h5>Insertions locations</h5>
              </div>
              <div class='card-body' style='max-height:80vh;overflow-y:auto'>
		        <div id="fig_summary_2.Insertions_quantification">
    
        
            <a href="2.Insertions_quantification.pdf"><img src="2.Insertions_quantification.png" width='70%'></a>
        
    
    <label class="labelpadding">Insertions location comparison for amplicon Reference; Top: percent difference; Bottom: p-value.</label>
    
        <p class="m-0"><small>Data: <a href="Insertions_quantification.txt">Insertions quantification</a></small></p>
    
    <br>
</div>
              </div>
            </div>
            
            <div class='card text-center mb-2'>
              <div class='card-header'>
                <h5>Deletions locations</h5>
              </div>
              <div class='card-body' style='max-height:80vh;overflow-y:auto'>
		        <div id="fig_summary_2.Deletions_quantification">
    
        
            <a href="2.Deletions_quantification.pdf"><img src="2.Deletions_quantification.png" width='70%'></a>
        
    
    <label class="labelpadding">Deletions location comparison for amplicon Reference; Top: percent difference; Bottom: p-value.</label>
    
        <p class="m-0"><small>Data: <a href="Deletions_quantification.txt">Deletions quantification</a></small></p>
    
    <br>
</div>
              </div>
            </div>
            
            <div class='card text-center mb-2'>
              <div class='card-header'>
                <h5>Substitutions locations</h5>
              </div>
              <div class='card-body' style='max-height:80vh;overflow-y:auto'>
		        <div id="fig_summary_2.Substitutions_quantification">
    
        
            <a href="2.Substitutions_quantification.pdf"><img src="2.Substitutions_quantification.png" width='70%'></a>
        
    
    <label class="labelpadding">Substitutions location comparison for amplicon Reference; Top: percent difference; Bottom: p-value.</label>
    
        <p class="m-0"><small>Data: <a href="Substitutions_quantification.txt">Substitutions quantification</a></small></p>
    
    <br>
</div>
              </div>
            </div>
            
            <div class='card text-center mb-2'>
              <div class='card-header'>
                <h5>Combined modifications (insertions, deletions and substitutions) locations</h5>
              </div>
              <div class='card-body' style='max-height:80vh;overflow-y:auto'>
		        <div id="fig_summary_2.All_modifications_quantification">
    
        
            <a href="2.All_modifications_quantification.pdf"><img src="2.All_modifications_quantification.png" width='70%'></a>
        
    
    <label class="labelpadding">Combined modifications (insertions, deletions and substitutions) location comparison for amplicon Reference; Top: percent difference; Bottom: p-value.</label>
    
        <p class="m-0"><small>Data: <a href="All_modifications_quantification.txt">Combined modifications (insertions, deletions and substitutions) quantification</a></small></p>
    
    <br>
</div>
              </div>
            </div>
            
            <div class='card text-center mb-2'>
              <div class='card-header'>
                <h5>Alleles enriched in Cas9</h5>
              </div>
              <div class='card-body' style='max-height:80vh;overflow-y:auto'>
		        <div id="fig_summary_3.Cas9.Alleles_frequency_table_around_sgRNA_GGAATCCCTTCTGCAGCACC_top">
    
        
            <a href="3.Cas9.Alleles_frequency_table_around_sgRNA_GGAATCCCTTCTGCAGCACC_top.pdf"><img src="3.Cas9.Alleles_frequency_table_around_sgRNA_GGAATCCCTTCTGCAGCACC_top.png" width='70%'></a>
        
    
    <label class="labelpadding">Distribution comparison of alleles. Nucleotides are indicated by unique colors (A = green; C = red; G = yellow; T = purple). Substitutions are shown in bold font. Red rectangles highlight inserted sequences. Horizontal dashed lines indicate deleted sequences. The vertical dashed line indicates the predicted cleavage site. The proportion and number of reads is shown for each sample on the right, with the values for Cas9 followed by the values for Untreated. Alleles are sorted for enrichment in Cas9.</label>
    
        <p class="m-0"><small>Data: <a href="Cas9.Alleles_frequency_table_around_sgRNA_GGAATCCCTTCTGCAGCACC.txt">Allele comparison table</a></small></p>
    
    <br>
</div>
              </div>
            </div>
            
            <div class='card text-center mb-2'>
              <div class='card-header'>
                <h5>Alleles enriched in Untreated</h5>
              </div>
              <div class='card-body' style='max-height:80vh;overflow-y:auto'>
		        <div id="fig_summary_3.Cas9.Alleles_frequency_table_around_sgRNA_GGAATCCCTTCTGCAGCACC_bottom">
    
        
            <a href="3.Cas9.Alleles_frequency_table_around_sgRNA_GGAATCCCTTCTGCAGCACC_bottom.pdf"><img src="3.Cas9.Alleles_frequency_table_around_sgRNA_GGAATCCCTTCTGCAGCACC_bottom.png" width='70%'></a>
        
    
    <label class="labelpadding">Distribution comparison of alleles. Nucleotides are indicated by unique colors (A = green; C = red; G = yellow; T = purple). Substitutions are shown in bold font. Red rectangles highlight inserted sequences. Horizontal dashed lines indicate deleted sequences. The vertical dashed line indicates the predicted cleavage site. The proportion and number of reads is shown for each sample on the right, with the values for Cas9 followed by the values for Untreated. Alleles are sorted for enrichment in Untreated.</label>
    
        <p class="m-0"><small>Data: <a href="Cas9.Alleles_frequency_table_around_sgRNA_GGAATCCCTTCTGCAGCACC.txt">Allele comparison table</a></small></p>
    
    <br>
</div>
              </div>
            </div>
            
          

          

            <div align="center" class='p-3'>
  
  <button class='btn btn-primary hidden-print' onclick='window.print();'><i class="fas fa-print"></i> Print</button>
</div>

	</div>  <!-- end jumbotron_content -->
</div>  <!-- end jumbotron -->

</div>  <!-- end column -->

<div class="col-sm-1"></div>
</div>


      
      <div class="row">
      <div class="col-sm-1"></div>
      <footer class="col-sm-10 crispresso_body_div screen-only py-5">
        <p class="text-center">
      If you like CRISPResso2 please support us by citing it in your work: <br>
          Clement K, Rees H, Canver MC, Gehrke JM, Farouni R, Hsu JY, Cole MA, Liu DR, Joung JK, Bauer DE, Pinello L.<br>
          <a href="https://rdcu.be/boxVG" target="_blank">CRISPResso2 provides accurate and rapid genome editing sequence analysis. </a><br>
          Nat Biotechnol. 2019 Mar; 37(3):224-226. doi: 10.1038/s41587-019-0032-3. PubMed PMID: 30809026.
        </p>
        <p class="text-center"><small> &copy; Copyright <a href="http://pinellolab.org">Kendell Clement and Luca Pinello</a></small></p>
        <p class="text-center">
          <small>
        <a href="https://twitter.com/intent/tweet?text=CRISPResso2%20provides%20accurate%20and%20rapid%20genome%20editing%20sequence%20analysis%20%40kendellclement%20%40lucapinello%20%23CRISPR%20%23CRISPResso2"
          class="twitter-share-button" data-via="kendellclement" data-size="large" data-count="none" data-hashtags="CRISPR" data-url="http://crispresso2.pinellolab.org">Tweet about CRISPresso2!</a>
        </small>
        </p>

      </footer>
      <div class="col-sm-1"></div>
    </div>
    
    </div>
    

  </body>

</html>